src/Testing/BaseMutatorTestCase.php

```diff
@@ -179,19 +180,25 @@ private function getMutationsFromCode(string $code, array $settings): array
 
     private function assertSyntaxIsValid(string $realMutatedCode): void
     {
-        exec(
-            sprintf('echo %s | php -l', escapeshellarg($realMutatedCode)),
-            $output,
-            $returnCode,
-        );
-
-        $this->assertSame(
-            0,
-            $returnCode,
-            sprintf(
-                'Mutator %s produces invalid code',
-                $this->mutator->getName(),
-            ),
-        );
+        try {
+            $tokens = token_get_all($realMutatedCode, TOKEN_PARSE);
+
+            $this->assertTrue(
+                $tokens !== [],
+                sprintf(
+                    'Mutator %s produces invalid code: %s',
+                    $this->mutator->getName(),
+                    $realMutatedCode,
+                ),
+            );
```

Comment:
this assertion is actually always true, but only invoking `token_get_all($realMutatedCode, TOKEN_PARSE)` without checking the result makes PHPStorm error as it thinks the `token_get_all` call is useless.